lib/engine/step/tracker.rb

```diff
@@ -11,6 +11,9 @@ def setup
       end
 
       def can_lay_tile?(entity)
+        step_name = self.class.name.split('::').last
+        return true if tile_lay_abilities(entity, time: step_name, allow_passive: false)
```

Comment:
This allows for abilities to generally block a Track-laying step from moving on (see https://github.com/tobymao/18xx/pull/2947/files#r550727703)